src/Application/Matching/OptimisticRedisAdapter.php

```diff
@@ -19,26 +20,30 @@
  */
 class OptimisticRedisAdapter extends Adapter
 {
-    private EntityManagerInterface $entityManager;
-    /** @var CampaignFunding[] */
-    private array $fundingsToPersist = [];
-    /** @var int Number of times to immediately try to allocate a smaller amount if the fund's running low */
-    private int $maxPartialAllocateTries = 5;
-    private Redis $redis;
     /**
-     * @var int How many seconds the authoritative source for real-time match funds should keep data, as a minimum.
-     *          Because Redis sets an updated value on each change to the balance, the case where using the database
-     *          value could be problematic (race conditions with high volume access) should not overlap with the case
-     *          where Redis copies of available fund balances are expired and have to be re-fetched.
+     * @param Redis $redis
+     * @param EntityManagerInterface $entityManager
+     * @param array $fundingsToPersist
+     * @param int $maxPartialAllocateTries -
+     *            How many seconds the authoritative source for real-time match funds should keep data, as a minimum.
+     *            Because Redis sets an updated value on each change to the balance, the case where using the database
+     *            value could be problematic (race conditions with high volume access) should not overlap with the case
+     *            where Redis copies of available fund balances are expired and have to be re-fetched.
+     *
+     * @param int $storageDurationSeconds
+     *            Number of times to immediately try to allocate a smaller amount if the fund's running low
      */
-    private static int $storageDurationSeconds = 86_400; // 1 day
-
-    public function __construct(Redis $redis, EntityManagerInterface $entityManager)
-    {
-        $this->redis = $redis;
-        $this->entityManager = $entityManager;
+    #[Pure]
+    public function __construct(
+        private Redis $redis,
+        private EntityManagerInterface $entityManager,
+        private array $fundingsToPersist = [],
```

Comment:
I'm not sure we make anything clearer by moving these 3 to constructor args, given no code instantiates the class passing them in.